navbar_alerts: Add upload file storage space warning banner.

sayamsamal opened the pull request:
This PR adds a navbar banner to warn users when their organization is approaching its file upload storage quota. The threshold for showing this banner is when the usage reaches 90% and 95% for admins and non-admins respectively.

For organizations on Cloud Free, the banner also includes a message about upgrading for more storage.

For admins of Cloud Free organizations, the banner also includes a link to the "/upgrade" page.

Fixes: #34441.

Follow-up PR for completion candidate #34516.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Description | Banner |
|--------|--------|
| Normal User in Cloud Free | <img width="1596" height="140" alt="Screenshot 2025-09-20 at 12 16 03 AM" src="https://github.com/user-attachments/assets/dbb29d96-2092-4e50-9fd9-4102a297805c" /> |
| Normal User not in Cloud Free | <img width="1596" height="140" alt="Screenshot 2025-09-20 at 12 36 49 AM" src="https://github.com/user-attachments/assets/240331a7-2037-4516-b49f-60133410f280" /> |
| Admin in Cloud Free | <img width="1596" height="140" alt="Screenshot 2025-09-20 at 12 14 34 AM" src="https://github.com/user-attachments/assets/a286858a-9b4a-4d31-b55e-1d75854f8c39" /> |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/navbar_alerts.ts:
```diff
@@ -124,6 +129,12 @@ export function set_last_upgrade_nag_dismissal_time(ls: LocalStorage): void {
 
+export function set_last_upload_quota_limit_reminder_time(ls: LocalStorage): void {
+    if (localstorage.supported()) {
+        ls.set("lastUpgradeQuotaLimitReminderTime", Date.now());
```
I guess it matches `lastUpgradeNagDismissalTime`.

Thinking about this, I think it might be a good idea to rework how the `ls` API works in order to have a single location where we declare the keys that are used, so that it's easy to see the prior art.

sayamsamal replied:
Yes, I tried to match the `lastUpgradeNagDismissalTime` naming convention. Do we prefer snakecase over here instead?

timabbott commented on web/src/navbar_alerts.ts:
```diff
@@ -173,6 +184,31 @@ export function toggle_organization_profile_incomplete_banner(): void {
+        new Date(last_upload_quota_banner_dismissal_time),
+        7,
+    ).getTime();
+    return Date.now() > upload_quota_banner_snooze_duration;
```
The `upload_quota_banner_snooze_duration` variable name is bad; it's not a duration, but instead the time after that duration. Maybe `upload_quota_banner_min_date` would be fine.

sayamsamal committed:
api: Add realm_upload_quota_used_bytes field to POST /register response.

This commit adds a new field `realm_upload_quota_used_bytes` to the
response of the POST /register API endpoint.

This field provides information about the total bytes of upload space
currently used by the realm, which when used in combination with the
`realm_upload_quota_mib` field, can be useful for clients to display
the upload quota usage to users.

Fixes #36648.

sayamsamal committed:
navbar_alerts: Add org approaching upload quota limit warning banner.

This commit adds a navbar banner to warn users when their organization's
file upload usage is approaching the org's upload quota limit, if any.

The threshold for showing this banner is when the usage reaches 90% and
95% for admins and non-admins respectively. The different thresholds are
in place to give admins a head start to take the required action before
nudging non-admin users with the warning banner.

For organizations on Cloud Free, the banner includes a message about
upgrading for more storage.

For admins of Cloud Free organizations, the banner also includes a link
to the "/upgrade" page.

Fixes #34441.
